### PR TITLE
all: add docs

### PR DESCRIPTION
**Pull request by mewmew**
**DO NOT MERGE**

This commit adds basic documentation to
all global variables that have been
documented in the sanctuary/notes
repo.

Note, this commit is entirely automated,
so I would expect it to have a lot of
issues.

This PR is simply to start the process
of adding documentation to the Devilution
code base. It would make sense to add this
to diasurgical/devilution rather than
diasurgical/devilutionX, but for now it was
easier to get running on devilutionX (mainly
since it handles missing windows.h include
header, etc).

We should probably not merge this PR as is,
but rather start discussing and fixing the
docs that are bad/missing. There are quite
a few "TODO: add docs" in there.

After these discussions, I'll merge the
updated docs into sanctuary/notes and then
update this PR automatically using the same
tools as was used to create it.

For reference, the tools used are as follows:

In the sanctuary/notes repo:

	cdoc2json -clang_args="-m32|-I./include|-I/usr/lib/clang/8.0.1/include" rdata/*.h data/*.h bss/*.h

In the diasurgical/devilutionX repo:

	addcdocs -clang_args="-D DEVILUTION_STUB|-D DEVILUTION_ENGINE|-I./SourceS|-I/usr/lib/clang/8.0.1/include" Source/*.cpp

The code of these two tools is ugly. I would
be surprised if anyone managed to get them to
run on any other machine than my laptop, simply
because it took quite some working getting the
Clang environment running. Still, for the
morbidly curious the tools are hosted here:

	https://github.com/mewspring/cdoc2json

### CODE REVIEW

**Comment by AJenbo** [comment]
Could we change it to use comment blocks instead of multiple line comments?

Also possibly use some relevant properties, or even just make up ower own ` * @address 0x4B7A34`

File: Source/player.cpp [hunk 1]
```diff
@@ -19,41 +19,161 @@ int plr_sframe_size;
...
 const char ArmourChar[4] = { 'L', 'M', 'H', 0 };
+// address: 0x47F20C
+//
+// WepChar maps from weapon animation to letter used in graphic files.
+//
+// PSX ref (SLPS-01416): 0x80110B68
+// PSX def: char WepChar[10]
+// Note: char[9] is enough, though PSX details as char[10].
+// alias: player_weapons
 const char WepChar[10] = { 'N', 'U', 'S', 'D', 'B', 'A', 'M', 'H', 'T', 0 };
```
Comment (AJenbo) [inline Source/player.cpp]: 4 is used since the loops continue until they hit a null.
`for (w = &WepChar[0]; *w; w++) {`

Comment (mewmew) [reply Source/player.cpp]: Ah, that makes sense. Kind of which they did:

```cpp
for (i = 0; i < sizeof(WepChar)/sizeof(WepChar[0]); i++)
```

**Comment by mewmew** [comment]
> Could we change it to use comment blocks instead of multiple line comments?

Sounds good. We can use multi-line comments instead.

> Also possibly use some relevant properties, or even just make up ower own * @address 0x4B7A34

Sure.

Lets make a list of properties we wish for the comments to have, then I can re-submit the PR.

I thought of one that becomes more relevant as Hellfire (and perhaps other versions of Diablo) are added. We should specify what version of Diablo the `address` comment refers to.

```diff
-address: 0xDEADBEEF
+address (v1.09b): 0xDEADBEEF
```

* Diablo 1.09b: `v1.09b`
* Hellfire 1.01: `...`

Exactly what format to use for these we can discuss and just decide upon one.

**Comment by mewmew** [comment]
> Could we change it to use comment blocks instead of multiple line comments?

Done.

**Comment by mewmew** [comment]
> Also possibly use some relevant properties, or even just make up ower own * @address 0x4B7A34

Are these properties for doxygen? We can easily add properties like these, so long as we standardize upon their format.

**Comment by AJenbo** [comment]
Manually plucked the doc from 26 of the files, just 47 left to go.

**Comment by AJenbo** [comment]
Plucked docs from all files with only a few comments, now down to 30 left.

**Comment by AJenbo** [comment]
automap.cpp, control.cpp and diablo.cpp notes have been plucked in this PR:
https://github.com/diasurgical/devilution/pull/2026

**Comment by mewmew** [comment]
> Also, Devilution has been merged so now the documentaiton that i plucked previously can also be found in this repo. Please don't merge or try to update this PR as it will mess up the view where i can track what i still need to pulick :)

No worries, I'll leave it as is. Great that you are integrating the documentation!

**Comment by AJenbo** [comment]
All remaning documentation, except the 4 drlg_l*.cpp files, are now part of this PR https://github.com/diasurgical/devilution/pull/2057.

**Comment by AJenbo** [comment]
All documentation has now been plucked :D

**Comment by mewmew** [comment]
great work @AJenbo!

**Pull request closed without merging**